constructor/construct.py

```diff
@@ -309,6 +333,20 @@
 ]
 
 
+_EXTRA_ENVS_SCHEMA = {
+    "specs": (list, tuple),
+    "environment": (str,),
+    "environment_file": (str,),
+    "channels": (list, tuple),
+    "channels_remap": (list, tuple),
+    "user_requested_specs": (list, tuple),
+    "exclude": (list, tuple),
+    # TODO: we can't support menu_packages for extra envs yet
+    # will implement when the PR for new menuinst lands
+    # "menu_packages": (list, tuple),
```

Comment:
Would this require https://github.com/conda/constructor/pull/474 (and thus https://github.com/conda/conda/pull/11035 and https://github.com/conda/menuinst/pull/91)? I'm having a bit of trouble following... I guess all of https://github.com/conda-incubator/ceps/pull/8 should be merged around the same time to enable "new menuinst"?

Reply:
Yep. The new `menuinst` support will be trickier to land, as it requires sync across many projects.